internal/daemon/daemon.go

```diff
@@ -472,6 +491,20 @@ func (d *Daemon) Start() {
 		return nil
 	})
 
+	if d.httpListener != nil {
+		// Start regular HTTP API for handling URLs like /v1/health
+		httpAPI := httpapi.NewAPI(d.overlord.CheckManager())
+		d.httpServer = &http.Server{Handler: httpAPI}
+		d.tomb.Go(func() error {
+			logger.Debugf("Starting HTTP API on http://localhost%s", d.httpServer.Addr)
```

Comment:
This is not true I believe. :4000 is the same as 0.0.0.0:4000, which is not the same as 127.0.0.1:4000.

I'm also slightly concerned about the introduction of this logic without us having discussed what we want in the medium term.  Per conversations we had too many months ago (you were probably not around, Ben), Pebble was designed to eventually support the same protocol over other transports in addition to its unix socket.  It feels like this is bolted in quickly and without much thinking around how to name this API, or how to integrate this server, in a way that faciliates that work in the near future.

To be clear, I'm not suggesting we need to do much work now to support that future reality, but we should at least understand how we want to design its entry points so that we can move there when the time comes.

Reply:
Yeah, fair enough. See comments below.